use_case_examples/cifar_brevitas_training/trainer.py

```diff
@@ -36,20 +37,57 @@
 from torchvision.datasets import CIFAR10
 
 
-def accuracy(output, target, topk=(1,)):
-    """Computes the precision@k for the specified values of k"""
-    maxk = max(topk)
-    batch_size = target.size(0)
+def accuracy(
+    output: torch.Tensor, target: torch.Tensor, topk: Tuple[int] = (1,)
+) -> List[torch.FloatTensor]:
+    """Computes the accuracy over the k top predictions for the specified values of k.
 
-    _, pred = output.topk(maxk, 1, True, True)
-    pred = pred.t()
-    correct = pred.eq(target.view(1, -1).expand_as(pred))
+    In top-5 accuracy you give yourself credit for having the right answer if the right answer
+    appears in your top five guesses.
+    Taken from https://discuss.pytorch.org/t/top-k-error-calculation/48815/2.
 
-    res = []
-    for k in topk:
-        correct_k = correct[:k].flatten().float().sum(0)
-        res.append(correct_k.mul_(100.0 / batch_size))
-    return res
+    Args:
+        output (torch.Tensor): The prediction of the model (scores, logits, raw y_pred) before
+            normalization or getting classes
+        target (torch.Tensor) : The target data representing the truth.
+        topk (Tuple[int]): Tuple of topk's to compute top 1, top 2 and top 5.
+
+    Returns:
+      list_topk_accs (List[torch.FloatTensor]): The list of topk accuracies [top1st, top2nd, ...]
+        depending on the topk input.
+    """
+    with torch.no_grad():
+        # Get the largest k and batch size
+        max_k = max(topk)
+        batch_size = target.size(0)
+
+        # Get top max_k indices that correspond to the most likely probability scores
+        _, y_pred = output.topk(k=max_k, dim=1)
+
+        # Transpose it to shape [max_k, batch_size]
+        y_pred = y_pred.t()
+
+        # Get the credit for each example if the models predictions is in max_k values
+        target_reshaped = target.view(1, -1).expand_as(y_pred)
+
+        # Compare every topk's model prediction with the ground truth
+        correct = y_pred == target_reshaped
+
+        # Compute topk accuracies
+        list_topk_accs = []
+        for k in topk:
+            # Get topk answers as a float tensor (1.0 and 0.0)
+            # The clone is necessary here in order to avoid miss-representations
+            topk_matched_truth = correct[:k].reshape(-1).clone().float()
```

Comment:
basically this was the issue with the previous `accuracy` function : for some obscure reasons, a `clone` is needed here 9else, the top_1 accuracy would not be right when calling `accuracy(..., topk=(1, 5))`